CMakeLists.txt

```diff
@@ -45,96 +76,75 @@ set(COMMON_WARNINGS
         -Wsign-compare
         -Wunused-parameter
         -Wpedantic
-        #-pedantic-errors
         -Wconversion
         -Wsign-conversion
         -Wreturn-type
         )
 # Clang-specific warnings compiler flags.
-set(CLANG_WARNINGS
+set(MATA_CLANG_WARNINGS
         -Winconsistent-missing-override
         -Wgnu-anonymous-struct
         -Wnested-anon-types
         -Wno-system-headers
         )
 
 # Set whether to treat warnings as errors.
-option(WERROR "WERROR" OFF)
-message("-- Treating warnings as errors: ${WERROR}")
-if (WERROR)
-    set(COMMON_WARNINGS "${COMMON_WARNINGS}" -Werror)
+if (MATA_WERROR)
+    message("-- Treating warnings in mata as errors")
+    set(MATA_COMMON_WARNINGS "${MATA_COMMON_WARNINGS}" -Werror)
 endif ()
 
-##############################################################################
-#                                COVERAGE
-##############################################################################
-
 # if DEBUG, also test coverage
-if(CMAKE_BUILD_TYPE MATCHES "Coverage")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g -O0 -fprofile-arcs -ftest-coverage")
```

Comment:
Why did hyou remove these flags? The `-fprofile-arcs` and `-ftest-coverage` should be necessary for testing coverage. At least locally.

Reply:
Both local and GA coverage seems to work, so... It should be fine, it seems. I will retest it again before merging.

Reply:
I replaced it with `--coverage`, which should do [the same thing](https://stackoverflow.com/questions/28840261/gcov-what-is-the-difference-between-coverage-and-ftest-coverage-when-buildi). Can you check if it works locally for you?

Reply:
Ook, that is neat, I will believe you. Codecov works and that is sufficient.

Reply:
I confirm that it works for me locally, and Codecov is not complaining here, either.